### PR TITLE
feat: add membership cronjob(garbage collector, stack cycles) and job(init nats publisher)

### PR DESCRIPTION
**Pull request by Dav-14**

### CODE REVIEW

File: charts/cloudprem/README.md [hunk 1]
```diff
@@ -425,6 +426,12 @@ Dex:
 
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
+| global.nats.auth.existingSecret | string | `""` |  |
+| global.nats.auth.password | string | `nil` |  |
+| global.nats.auth.secretKeys.password | string | `"password"` |  |
+| global.nats.auth.secretKeys.username | string | `"username"` |  |
+| global.nats.auth.user | string | `nil` |  |
+| global.nats.enabled | bool | `false` |  |
```
Comment (Dav-14) [inline charts/cloudprem/README.md:434]: Adding nats global values

File: charts/cloudprem/README.md [hunk 2]
```diff
@@ -476,12 +483,15 @@ Dex:
...
+| membership.config.auth.additionalOAuthClients | list | `[]` | Membership additional oauth clients |
 | membership.config.auth.tokenValidity | object | `{"accessToken":"5m","refreshToken":"72h"}` | According to "nsuµmh" And https://github.com/spf13/cast/blob/e9ba3ce83919192b29c67da5bec158ce024fdcdb/caste.go#L61C3-L61C3 |
 | membership.config.fctl | bool | `true` | Enable Fctl |
 | membership.config.grpc.existingSecret | string | `""` |  |
 | membership.config.grpc.secretKeys.secret | string | `"TOKENS"` |  |
-| membership.config.grpc.tokens | list | `["changeMe"]` | Membership agent grpc token |
+| membership.config.grpc.tokens | list | `[]` | Membership agent grpc token |
+| membership.config.job | object | `{"garbageCollector":{"concurrencyPolicy":"Forbid","enabled":false,"resources":{},"restartPolicy":"Never","schedule":"0 0 * * *","startingDeadlineSeconds":200,"suspend":false,"tolerations":[]},"stackLifeCycle":{"additionalEnv":[],"concurrencyPolicy":"Forbid","enabled":false,"resources":{},"restartPolicy":"Never","schedule":"*/30 * * * *","startingDeadlineSeconds":200,"suspend":false,"tolerations":[]}}` | CronJob to manage the stack life cycle and the garbage collector |
```
Comment (Dav-14) [inline charts/cloudprem/README.md:492]: adds the job config

File: charts/cloudprem/README.md [hunk 3]
```diff
@@ -492,6 +502,20 @@ Dex:
 | membership.config.oidc.scopes | list | `["openid","email","federated:id"]` | Membership oidc redirect uri |
 | membership.config.oidc.scopes[2] | string | `"federated:id"` | Membership Dex federated id scope |
 | membership.config.oidc.secretKeys | object | `{"secret":""}` | Membership oidc secret key |
+| membership.config.publisher.clientID | string | `"membership"` |  |
+| membership.config.publisher.jetstream.replicas | int | `1` |  |
+| membership.config.publisher.topicMapping | string | `"membership"` |  |
```
Comment (Dav-14) [inline charts/cloudprem/README.md:507]: Publisher client related values

File: charts/cloudprem/README.md [hunk 4]
```diff
@@ -492,6 +502,20 @@ Dex:
...
+| membership.config.stack.cycle.delay.disable | string | `"72h"` |  |
+| membership.config.stack.cycle.delay.disablePollingDelay | string | `"1m"` |  |
+| membership.config.stack.cycle.delay.disposable | string | `"360h"` |  |
+| membership.config.stack.cycle.delay.prune | string | `"720h"` |  |
+| membership.config.stack.cycle.delay.prunePollingDelay | string | `"1m"` |  |
+| membership.config.stack.cycle.delay.warn | string | `"72h"` |  |
+| membership.config.stack.cycle.dryRun | bool | `true` |  |
```
Comment (Dav-14) [inline charts/cloudprem/README.md:514]: Stack Cycle related config

File: charts/cloudprem/README.md [hunk 5]
```diff
@@ -492,6 +502,20 @@ Dex:
...
+| membership.config.stack.minimalStackModules[0] | string | `"Auth"` |  |
+| membership.config.stack.minimalStackModules[1] | string | `"Ledger"` |  |
+| membership.config.stack.minimalStackModules[2] | string | `"Payments"` |  |
+| membership.config.stack.minimalStackModules[3] | string | `"Gateway"` |  |
```
Comment (Dav-14) [inline charts/cloudprem/README.md:518]: Minimal modules set within a stack

File: charts/core/templates/_broker.tpl [hunk 6]
```diff
@@ -17,15 +17,32 @@
...
-  value: "{{ .Values.config.nats.clientID}}"
+  value: "{{ .Values.config.publisher.clientID}}"
 - name: PUBLISHER_TOPIC_MAPPING
-  value: "{{ .Values.config.nats.topicMapping }}"
-{{- end -}}
-{{- end -}}
\ No newline at end of file
+  value: "{{ .Values.config.publisher.topicMapping }}"
+{{- end }}
+{{- end }}
```
Comment (Dav-14) [inline charts/core/templates/_broker.tpl:48]: Add Authentification

File: charts/membership/templates/cronjob-gc.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
+{{- if .Values.config.job.garbageCollector.enabled }}
```
Comment (Dav-14) [inline charts/membership/templates/cronjob-gc.yaml:1]: DIsabled by default

File: charts/membership/templates/cronjob-stack-lifeycle.yaml [hunk 8]
```diff
@@ -0,0 +1,63 @@
+{{- if .Values.config.job.stackLifeCycle.enabled }}
```
Comment (Dav-14) [inline charts/membership/templates/cronjob-stack-lifeycle.yaml:1]: disabled by default

File: charts/membership/values.yaml [hunk 9]
```diff
@@ -338,12 +353,95 @@ config:
...
+
+    # Modules created by default on a stack
+    minimalStackModules:
+      - Auth
+      - Ledger
+      - Payments
+      - Gateway
+
+    cycle:
+      dryRun: true
```
Comment (Dav-14) [inline charts/membership/values.yaml:381]: DryRun by default

File: charts/membership/templates/cronjob-stack-lifeycle.yaml [hunk 10]
```diff
@@ -0,0 +1,63 @@
+{{- if and .Values.config.job.stackLifeCycle.enabled .Values.global.nats.enabled }}
```
Comment (Dav-14) [inline charts/membership/templates/cronjob-stack-lifeycle.yaml:1]: Does not run if nats is disabled

File: charts/membership/templates/job.yaml [hunk 11]
```diff
@@ -28,3 +28,17 @@ spec:
...
+          {{- include "membership.image" . | nindent 10 }}
+          args:
+            - stream
+            - init
+          env:
+          - name: JETSTREAM_REPLICAS
+            value: "{{ .Values.config.publisher.jetstream.replicas }}"
+          {{- include "core.env.common" . | nindent 10 }}
+          {{- include "core.nats.env" . | nindent 10 }}
+        {{- end }}
```
Comment (Dav-14) [inline charts/membership/templates/job.yaml:44]: Does not init if nats is disabled

**Review by flemzord — APPROVED** [review]

**Pull request merged**